azext_iot/iothub/providers/state.py

```diff
@@ -155,6 +155,8 @@ def migrate_state(
             raise MutuallyExclusiveArgumentError(usr_msgs.LOGIN_WITH_ARM_ERROR)
         if HubAspects.Arm.value not in hub_aspects and not self.target:
             raise ResourceNotFoundError(usr_msgs.TARGET_HUB_NOT_FOUND_MSG.format(self.hub_name))
+        if not self.rg and not self.target:
+            self.rg = orig_hub_target.get("resourcegroup")
```

Comment:
there is a possibility of using the connection string for the original hub target - which will result in a target with no resource group. The command should fail gracefully before that, but I will keep this get just in case